src/cljs/harja/views/hallinta/urakkatiedot/paallystysilmoitukset_nakyma.cljs

```diff
@@ -0,0 +1,241 @@
+
+    :else
+    [:span "Ei aloitettu"]))
+
+(defn- lahetys-yha-velho-nappi [e! {:keys [oikeus urakka-id sopimus-id vuosi paallystysilmoitus kohteet-yha-velho-lahetyksessa valittu-urakka]}]
+  (let [kohde-id (:paallystyskohde-id paallystysilmoitus)
+        kun-onnistuu-fn #(e! (tiedot/->ValitseUrakka %))]
+    [napit/palvelinkutsu-nappi
+     "Lähetä"
+     #(do
+        (log "[YHA/VELHO] Lähetetään urakan (id:" urakka-id ") sopimuksen (id: " sopimus-id
+          ") kohde (id:" (pr-str kohde-id) ") YHA:n")
+        (k/post! :laheta-pot-yhaan-ja-velhoon {:urakka-id urakka-id
+                                               :sopimus-id sopimus-id
+                                               :kohde-id kohde-id
+                                               :vuosi vuosi}
+          nil
+          true))
+     {:ikoni (ikonit/envelope)
+      :luokka :napiton-nappi
+      :disabled (or false
+                  (not (oikeudet/on-muu-oikeus? "sido" oikeus urakka-id @istunto/kayttaja)))
+      :virheviestin-nayttoaika viesti/viestin-nayttoaika-pitka
+      :kun-valmis #(do) 
+      :kun-onnistuu (fn [vastaus]
+                      (kun-onnistuu-fn valittu-urakka)
+                      (log "[YHA/VELHO] Lähetys onnistui urakan (id:" urakka-id ") sopimuksen (id: " sopimus-id
+                        ") kohde (id:" (pr-str kohde-id) ") YHA:an. Vastaus: " (pr-str vastaus)))
+      :kun-virhe (fn [vastaus]
+                   (kun-onnistuu-fn valittu-urakka)
+                   (log "[YHA] Lähetys epäonnistui osalle kohteista YHAan. Vastaus: " (pr-str vastaus)))
+      :nayta-virheviesti? false}]))
+
+(defn- kaikki-lahetys-yha-velho-nappi [e! {:keys [oikeus urakka-id sopimus-id vuosi paallystysilmoitus valittu-urakka]}]
+  (let [ilmoituksen-voi-lahettaa? (fn [{:keys [paatos-tekninen-osa tila lahettaja] :as paallystysilmoitus}]
+                                    (and (= :hyvaksytty paatos-tekninen-osa)
+                                      (contains? #{:valmis :lukittu} tila)
+                                      (nil? lahettaja)))
+        kohde-id (map #(:paallystyskohde-id %) (filter ilmoituksen-voi-lahettaa? paallystysilmoitus))
+        kaikki-kohteet-maara (count paallystysilmoitus)
+        kohteet-lahetyksessa-maara (count kohde-id) 
+        kun-onnistuu-fn #(e! (tiedot/->ValitseUrakka %))
+        paivita-urakkatilanne-fn #(e! (tiedot/->HaePaallystysUrakat %))]
+    [napit/palvelinkutsu-nappi
+     (str "Kehittäjä: Lähetä kaikki valmiit kohteet YHA:aan " kohteet-lahetyksessa-maara "/" kaikki-kohteet-maara)
+     #(do
+        (log "[YHA/VELHO] Lähetetään urakan (id:" urakka-id ") sopimuksen (id: " sopimus-id
+          ") kohde (id:" (pr-str kohde-id) ") YHA:n ja Velhoon (VELHO DISABLED)")
+        (k/post! :laheta-pot-yhaan-ja-velhoon {:urakka-id urakka-id
+                                               :sopimus-id sopimus-id
+                                               :kohde-id kohde-id
```

Comment:
Tosin tässä olettaisin, että ihan yhtä kohdetta ollaan lähettämässä.

Reply:
Joo näin on tässä tapauksessa.